### DISCUSSION TITLE
There is some way to silence notifications?

### DISCUSSION THREAD

**Discussion by Zeioth** [nvim-java/nvim-java#141] [Q&A] [answered]
Hello we've been shipping nvim-java with [NormalNvim](https://github.com/NormalNvim/NormalNvim) for a few months with great success. Now we would like to enable DAP support.

## Question 1:
There is some way to silence notifications?
![screenshot_2024-03-21_03-15-41_054371948](https://github.com/nvim-java/nvim-java/assets/3357792/46329b5e-b69b-4bf0-b045-84c6cb78e1f6)

## Question 2
What's the command to create a java project capable of running on DAP?
![screenshot_2024-03-21_03-22-47_418523680](https://github.com/nvim-java/nvim-java/assets/3357792/899bd492-d1b3-4d5f-a733-ae42b6694ab3)
![screenshot_2024-03-21_03-22-50_842412074](https://github.com/nvim-java/nvim-java/assets/3357792/61a1aa4f-534c-4e3f-b03c-11cf9bdc90b2)

**Comment by s1n7ax** [comment, marked as answer]
@Zeioth It says Client 1 quit. That's not a good sign even though it's just a single file. I will try NormalNvim when I have time and update you on this.

You cannot suppress the "Configuring Dap" & "Dap configured" notification at the moment. I would accept configurable PR for that. But jdtls progress report can be done through the default handler in lspconfig (I don't remember what it is but it might be 'language/status'). 

DAP works on maven, gradle or projects created by eclipse. Creating projects is not covered in the project. Any maven, gradle project would work.

You can use maven or gradle CLI to generate projects.

For maven:
[https://gist.github.com/adojos/2aaf3abeee9422e1840235636dea9c06](https://www.youtube.com/redirect?event=comments&redir_token=QUFFLUhqa1RObHI3eTZ5bDJTYmxiZlVLMEM0SFlCVHJYUXxBQ3Jtc0trT3ZIVGRPU3RrSy1Tdi00ZzJRUGVMdXRVRzNJTXNEYUVtRTB6NXFueXgxdGxDakgtRU44SHRvcnpONFhMZHl5WTFIZUd2NXJaMGljR3BGLVVQa0VmT1o2X01BVFRVS0FEd2FNeVItdE1ydnVfem1xYw&q=https%3A%2F%2Fgist.github.com%2Fadojos%2F2aaf3abeee9422e1840235636dea9c06&stzid=UgxcCcFtQxvxbmleKs94AaABAg.A-m4BG0CAO6A-ngiJB696w)

For Gradle:
[https://docs.gradle.org/current/samples/sample_building_java_applications.html](https://www.youtube.com/redirect?event=comments&redir_token=QUFFLUhqbWlYY1JzZDFwWVhjMEZrMmlYVkYweFJiUm5sUXxBQ3Jtc0tuVlR4MTBiUmdFRXZ5ckhObW1aanNPQm9HM0NKU3NZYXhNaDlSQno3UFFzenJjbGxGcTI2c1BOWnAxRGUzOGNGcjZEa1dDU2VWTExfMnVpU09zbGlxYlhtek1lYmlVOEh2ajNEVHZSNldEUjNCOUlhVQ&q=https%3A%2F%2Fdocs.gradle.org%2Fcurrent%2Fsamples%2Fsample_building_java_applications.html&stzid=UgxcCcFtQxvxbmleKs94AaABAg.A-m4BG0CAO6A-ngiJB696w)

For Spring Boot:
[https://start.spring.io](https://www.youtube.com/redirect?event=comments&redir_token=QUFFLUhqbEZKVmpnQzlxekFXSWMtbHFWY0p1VUFtd3VSQXxBQ3Jtc0tsRDF5dE51Tzh0RnNVSDZLRk90bVIyNzcwbXJ2cU9vWnBaeTVmWjN3ZTB0blpGa1pJMXZrNjhTb1ZsTG5lcnBJMXY5YjVKcGxuUVI5WG16MjltUndaaGM2SVprdmJEM0R6NU51clZOdnpqamJjRFlHaw&q=https%3A%2F%2Fstart.spring.io%2F&stzid=UgxcCcFtQxvxbmleKs94AaABAg.A-m4BG0CAO6A-ngiJB696w)

**Comment by Zeioth** [comment]
Thank you so much for your answer. 

Don't worry about the first warning because I found I accidentally had two versions of jdtls installed at the same time and that was causing it:
![screenshot_2024-03-21_05-42-59_654278144](https://github.com/nvim-java/nvim-java/assets/3357792/2dca42d9-2760-447b-be89-897bbd243bab)
Consider that in case we are versioning jdtls on the nvim-java mason registry, that might cause the issue in the future though, every time a new version is released without us uninstalling the old version for the user.

About the PR I have it ready. I'm gonna go to bed, review it in the morning and ship it 👍

**Reply by s1n7ax** [reply]
Ah shit. Noted. Created this issue. I should fix it before next jdtls release
https://github.com/nvim-java/nvim-java/issues/143
